bodhi/server/validators.py

```diff
@@ -227,13 +227,11 @@ def validate_builds(request, **kwargs):
                                    'Cannot edit stable updates')
 
         for nvr in request.validated.get('builds', []):
-            # If the build is new
-            if nvr not in edited:
-                # Ensure it doesn't already exist
-                build = request.db.query(Build).filter_by(nvr=nvr).first()
-                if build and build.update is not None:
-                    request.errors.add('body', 'builds',
-                                       "Update for {} already exists".format(nvr))
+            # Ensure it doesn't already exist in another update
+            build = request.db.query(Build).filter_by(nvr=nvr).first()
+            if build and build.update is not None and up.title != build.update.title:
```

Comment:
Is the intention here to compare ```up``` to ```build.update```? If so, it might be good to compare the ```alias``` field instead of the ```title``` field since we want to remove/repurpose the ```title``` field some day.

Reply:
Also, I'm not sure, but you might be able to use the ```is``` operator to compare them. I'd have to read some sqlalchemy docs to verify. If it were me, I'd just play it safe and compare a field I know to be unique like you did here.

Reply:
Sure, I'll switch to the `.alias` field.
However, the whole problem is that sometimes the update's title doesn't get correctly updated, and on later update, if one of the builds in the update remain, it will abort otherwise.

My example from the test:
- I submitted `a-1.0-1.fc17 b-1.0-1.fc17 c-1.0-1.fc17`
- At some point, I edited that to replace `b-1.0-1.fc17` with `b-1.0-2.fc17`. However, due to a sometimes-occuring bug, the `update.title` is still `a-1.0-1.fc17 b-1.0-1.fc17 c-1.0-1.fc17`. This has happened a few times, and given that we're intending to remove/repurpose title, I didn't feel like fixing that.
- I then want to edit it to replace `c-1.0-1.fc17` with `c-2.0-1.fc17`. At this point, Bodhi checks each of the new builds whether they are in an existing update that is not the update I'm testing. It does that by testing against the `nvr in update.title`. HOWEVER, since the title didn't update correctly after the first edit, it no longer sees `b-1.0-2.fc17` as part of this update, but when it goes request that build, that build does have an update attached to it.
- My last edit now fails, because `b-1.0-2.fc17 is in another update`.

This patch fixes that problem by instead of checking whether all of the new builds are listed in the `update.title`, just look if the update the build is in is the one we're currently editing, thus fixing the example.

Reply:
That makes sense, thanks for the detailed explanation!